File: cpp/common_class/object/object.cpp
#include "object.h"

object::object() {
	this->id = 0;
	this->name = "no_name";
	this->pos = vector3(0,0,0);
	this->scale = vector3(1,1,1);
}

object::object(int id, std::string name) {
	this->id = id;
	this->name = name;
	this->pos = vector3(0,0,0);
	this->scale = vector3(1,1,1);
}

object::object(vector3 pos, vector3 scale) {
	this->name = "no_name";
	this->pos = pos;
	this->scale = scale;
}

object::object(int id, vector3 pos, vector3 scale) {
	this->id = id;
	this->name = "no_name";
	this->pos = pos;
	this->scale = scale;
}

object::object(int id, std::string name, vector3 pos, vector3 scale) {
	this->id = id;
	this->name = name;
	this->pos = pos;
	this->scale = scale;
}

object::~object() {

}

/*
int main() {
  object* o = new object();

  return 0;
}
*/

File: py/game/object.py
#!/usr/bin/py
  

class Object:

  (
	PLANE,
	POLYGON,
  ) = range(0,2)

  def __init__(self):
	self.id = 0
	self.name = "NO_NAME"
	self.renderType = 0
	self.x = self.y = self.z = 0
	self.vertices = []

  def destroy(self):
	print("destroy")

  def render(self):
	print("render")

class Object2D:
  def __init__(self, x, y, w, h):
	self.x = x
	self.y = y
	self.w = w
	self.h = h

  def create(self, x, y, w, h):
	self.x = x
	self.y = y
	self.w = w
	self.h = h

  def destroy(self):
	print("destroy")

  def render(self):
	print("render")

if __name__ == "__main__":
  obj = Object()
  obj.render()
